src/PublicKey.h

```diff
@@ -49,6 +50,8 @@ class PublicKey {
         case 6:
         case 7:
             return PublicKeyType::secp256k1Extended;
+        case 11:
```

Comment:
What are the restrictions of the publicKey.bytes[0] value? Is 11 a safe value for ed25519-blake2b?

Also, since there's now more than one ed25519 key type, it feels weird to have a big gap in the type byte value for ed25519-sha512 (1) and ed25519-blake2b (11). Would it be safe to change the ed25519-sha512 byte value from 1 to 10? That would mean we'd have: `publicKey.bytes[0] = 10` for ed25519-sha512 and `publicKey.bytes[0] = 11` for ed25519-blake2b.

Reply:
This value does not encode the hash function used. It only encodes the elliptic curve. If we want to store the hash function we should add a new field, but I don't think it's necessary.

Reply:
ed25519-sha512 and ed25519-blake2b produce different results when deriving public key from the same private key (and different signatures). Thus making a distinction between which ed25519 is being used is necessary.

Reply:
I see. In that case I'm not sure it makes sense to keep using the first byte to represent that. We should have a more robust mechanism like a separate enum property. It's kind of messy because some blockchains take that first byte in secp256k1 as part of the input to the hashing function. But we should maybe just prepend the byte for blockchains that need it.

Reply:
In the current PublicKey implementation it's pretty much unclear what the first byte is supposed to be exactly. It's kind of like a curve indicator, but then some curves have more than one value that is associated with them, etc.
Storing the curves/options as enums would be a lot clearer and if the magic byte is needed by some publicKey accepting function, then encoding that information at the call site would be definitely be a better approach.